### PR TITLE
Chemmaster TGUI

### PR DESCRIPTION
**Pull request by FluffyGhoster**
Ported the unbelievably 11 years old curse code of the ChemMaster from htmlUI to TGUI.

Chemmaster:
image

Condimaster:
image

Chemmaster with everything opened, a blood sample analysis done, dylovene in the internal storage, blood dylo and inap in the beaker, bottle and pill selection open
image

### CODE REVIEW

**Comment by FluffyGhoster** [comment]
!review

**Review by Arrow768 — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 1]
```diff
@@ -75,81 +77,141 @@
...
-			if(href_list["amount"])
-				var/rtype = text2path(href_list["remove"])
-				var/amount = Clamp((text2num(href_list["amount"])), 0, 200)
+					analysis["name"] = G.name
+					analysis["type"] = Gdata["blood_type"]
+					analysis["DNA"] = Gdata["blood_DNA"]
+				else if(params["name"] == "Close")
+					analysis = list()
+
+			// This is not a feature complete parity, but this is some unbelievably cursed code, perhaps we can do without
```
Comment (NonQueueingMatt) [inline code/modules/reagents/Chemistry-Machinery.dm:189]: if we can do without it, delete it

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 2]
```diff
@@ -16,18 +16,20 @@
 	layer = 2.9
 	clicksound = /singleton/sound_category/button_sound
 
-	var/beaker = null
+	var/obj/item/reagent_containers/glass/beaker = null
 	var/obj/item/storage/pill_bottle/loaded_pill_bottle = null
 	var/mode = TRUE
 	var/condi = 0
```
Comment (NonQueueingMatt) [inline code/modules/reagents/Chemistry-Machinery.dm:22]: If it's a condimaster, make it have the idris background instead of the zenghu one.

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/reagents/Chemistry-Machinery.dm (hunk 1 above)
Comment (FluffyGhoster) [reply code/modules/reagents/Chemistry-Machinery.dm:189]: can we do without it?

Comment (FluffyGhoster) [reply code/modules/reagents/Chemistry-Machinery.dm]: actually, power move: we can

File: code/modules/reagents/Chemistry-Machinery.dm (hunk 2 above)
Comment (FluffyGhoster) [reply code/modules/reagents/Chemistry-Machinery.dm:22]: done

**Review by NonQueueingMatt — APPROVED** [review]

**Pull request merged**